sources/optimizer/bcknd/device/cuda/mma_kernel.h

```diff
@@ -35,71 +35,132 @@
 #ifndef MMA_CUDA_KERNEL_H
 #define MMA_CUDA_KERNEL_H
 
+template <typename T>
+__global__ void delta_1dbeam_kernel(T* __restrict__ Delta,
+     const T L_total, const T Le, const int offset, const int n) {
+  int k = blockIdx.x * blockDim.x + threadIdx.x;
+  if (k >= n) return;
+
+  // Convert to 1-based indexing for the calculation
+  int idx = k + 1;
+
+  // Calculate first term: (L_total - Le*(offset+idx-1))^3
+  T x1 = L_total - Le * (T)(offset + idx - 1);
+  T term1 = x1 * x1 * x1;
+
+  // Calculate second term: (L_total - Le*(offset+idx))^3  
+  T x2 = L_total - Le * (T)(offset + idx);
+  T term2 = x2 * x2 * x2;
+
+  // Final result
+  Delta[k] = (term1 - term2) / (T)3.0;
+}
+
 template <typename T>
 __global__ void mma_Ljjxinv_kernel(T* __restrict__ Ljjxinv,
      const T* __restrict__ pjlambda, const T* __restrict__ qjlambda,
      const T* __restrict__ x, const T* __restrict__ low, const T* __restrict__ upp,
      const T* __restrict__ alpha, const T* __restrict__ beta,
      const int n) {
   int tj = blockIdx.x * blockDim.x + threadIdx.x;
-  if (tj < n) {
-    const T xt = x[tj];
-    T val = -1.0 / (2.0 * pjlambda[tj] / pow(upp[tj] - xt, 3) +
-                    2.0 * qjlambda[tj] / pow(xt - low[tj], 3));
-    // Remove the sensitivity for the active primal constraints
-    bool is_alpha = xt == alpha[tj];
-    bool is_beta  = xt == beta[tj];
-    Ljjxinv[tj] = (is_alpha || is_beta) ? T(0.0) : val;
-  }
-}
+  if (tj >= n) return;
+
+  // Load inputs into registers
+  const T xt    = x[tj];
+  const T low_t = low[tj];
```

Comment:
Please do not use `_t` as a postfix. It is very commonly used to distinguish
types from variables.
```suggestion
  const T low_j = low[tj];
```